### PR TITLE
review

### PR DESCRIPTION
**Pull request by quannb99**
## Related Tickets:
- [#29304](https://edu-redmine.sun-asterisk.vn/issues/29304)

## WHAT this PR does?
- Create review

## Notes Impacted Areas:
- N/A.

## Attachments:
- rating

### CODE REVIEW

**Comment by quannb99** [comment]
ready

File: app/Http/Controllers/ReviewController.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(ReviewFormRequest $request)
+    {
+        Review::create($request->all());
+        $user = User::findOrFail($request->user_id);
+        $user->books()->syncWithoutDetaching([$request->book_id => ['rating' => $request->rating]]);
+        $book = Book::findOrFail($request->book_id);
```
Comment (huyenttm-0901) [inline app/Http/Controllers/ReviewController.php]: viết dòng find book lên trc nhé.

File: app/Http/Controllers/BookController.php [hunk 2]
```diff
@@ -81,11 +83,15 @@ public function show($id)
...
+            $rated = config('default.rating');
+            if ($book->users()->where('user_id', Auth::user()->id)->exists()) {
+                $rated = $book->users()->firstWhere('user_id', Auth::user()->id)->pivot->rating;
+            }
         } catch (ModelNotFoundException $e) {
             return redirect()->route('books')->with('fail_status', trans('msg.find_fail'));
         }
 
-        return view('show_book', compact(['book', 'selectedCategories', 'lastestBook', 'randomBook', 'categories']));
+        return view('show_book', compact(['book', 'selectedCategories', 'lastestBook', 'randomBook', 'categories', 'rated']));
```
Comment (huyenttm-0901) [inline app/Http/Controllers/BookController.php]: liệt kê xuống dòng nhé em

File: app/Models/Book.php [hunk 3]
```diff
@@ -23,6 +23,6 @@ public function categories()
 
     public function users()
     {
-        return $this->belongsToMany(User::class);
+        return $this->belongsToMany(User::class)->withPivot('rating','status','favorite');
```
Comment (huyenttm-0901) [inline app/Models/Book.php]: convention

File: app/Models/User.php [hunk 4]
```diff
@@ -38,6 +38,6 @@ public function reviews()
 
     public function books()
     {
-        return $this->belongsToMany(Book::class);
+        return $this->belongsToMany(Book::class)->withPivot('rating','status','favorite');
```
Comment (huyenttm-0901) [inline app/Models/User.php]: tương tự

**Comment by quannb99** [comment]
ready

**Pull request merged**